### ISSUE TITLE
NullReferenceException in ParatroopersPower PlayLaunchSounds

### ISSUE THREAD

**Issue by dwwwj222** [OpenRA/OpenRA#19647]
## Issue Summary

I am working on several features on a map, for example dropping paratroopers after a delay at the start, if I let the map run normally the paratroopers drop fine but if I try again this time disabling visibility checks at the start then the game crashes

DESCRIPTION

## Reproduction

STEPS TO REPRODUCE THE ISSUE
Enable debug mode and then disable visibility checks

## Expected behavior

EXPECTED BEHAVIOR

## Screenshots / Screen recordings / Replays

Runtime Version: Mono 6.4.0.198 (2019-06/fe64a4765e6 Wed Sep 18 14:14:51 EDT 2019) CLR 4.0.30319.42000
Exception of type `System.NullReferenceException`: Object reference not set to an instance of an object
  at OpenRA.Mods.Common.Traits.SupportPower.PlayLaunchSounds () [0x00097] in <15ed584efd4c42b4a11c49d2c6cfb506>:0 
  at OpenRA.Mods.Common.Traits.ParatroopersPower+<>c__DisplayClass4_0.<SendParatroopers>b__3 (OpenRA.World w) [0x0000d] in <15ed584efd4c42b4a11c49d2c6cfb506>:0 
  at OpenRA.World.Tick () [0x001c5] in <b2929aaf82634213b72ac90bbb9d3dd9>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x001bc] in <b2929aaf82634213b72ac90bbb9d3dd9>:0 
  at OpenRA.Game.LogicTick () [0x0003e] in <b2929aaf82634213b72ac90bbb9d3dd9>:0 
  at OpenRA.Game.Loop () [0x000f1] in <b2929aaf82634213b72ac90bbb9d3dd9>:0 
  at OpenRA.Game.Run () [0x0003c] in <b2929aaf82634213b72ac90bbb9d3dd9>:0 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00010] in <b2929aaf82634213b72ac90bbb9d3dd9>:0 
  at OpenRA.Launcher.Program.Main (System.String[] args) [0x00044] in <2173314defb2480ab12d714bd69ebcd6>:0

**Comment by pchote** [comment]
Can you please attach the map that causes the crash?

**Comment by dwwwj222** [comment]
[https://easyupload.io/9e7m0r](url)

**Comment by dwwwj222** [comment]
Sorry this is the correct file with the crash the previous one. was incorrect
https://easyupload.io/i1l0a6

Thanks

**Comment by Mailaender** [comment]
That map has several problems:

```
Testing map: Melons Nuclear Forest 0.3
OpenRA.Utility(1,1): Warning: Actor type `proc` grants conditions that are not consumed: critical
OpenRA.Utility(1,1): Error: Actor type `nuker` consumes conditions that are not granted: disabled
OpenRA.Utility(1,1): Error: Actor type `husk` does not define a default visibility type!
OpenRA.Utility(1,1): Error: Actor type `shockdrop` does not define a default visibility type!
OpenRA.Utility(1,1): Error: Actor type `nuker` does not define a default visibility type!
OpenRA.Utility(1,1): Error: Actor husk is not constructible; failure: ActorInfo("husk") failed to initialize because of the following:
Missing:
OpenRA.Mods.Common.Traits.Render.RenderSpritesInfo 
OpenRA.Mods.Common.Traits.BodyOrientationInfo 
OpenRA.Traits.IHealthInfo 
Unresolved:
OpenRA.Mods.Common.Traits.Render.WithIdleOverlayInfo: { OpenRA.Mods.Common.Traits.Render.RenderSpritesInfo, OpenRA.Mods.Common.Traits.BodyOrientationInfo }
OpenRA.Mods.Common.Traits.ChangesHealthInfo: { OpenRA.Traits.IHealthInfo }

OpenRA.Utility(1,1): Error: OpenRA.Mods.Common.Lint.CheckUnknownTraitFields failed with exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at OpenRA.Mods.Common.Lint.CheckUnknownTraitFields.CheckActors(IEnumerable`1 actors, Action`1 emitError, ModData modData) in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Lint/CheckUnknownTraitFields.cs:line 75
   at OpenRA.Mods.Common.Lint.CheckUnknownTraitFields.CheckMapYaml(Action`1 emitError, ModData modData, IReadOnlyFileSystem fileSystem, MiniYaml ruleDefinitions) in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Lint/CheckUnknownTraitFields.cs:line 92
   at OpenRA.Mods.Common.Lint.CheckUnknownTraitFields.OpenRA.Mods.Common.Lint.ILintMapPass.Run(Action`1 emitError, Action`1 emitWarning, ModData modData, Map map) in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Lint/CheckUnknownTraitFields.cs:line 30
   at OpenRA.Mods.Common.UtilityCommands.CheckYaml.OpenRA.IUtilityCommand.Run(Utility utility, String[] args) in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/UtilityCommands/CheckYaml.cs:line 108
Errors: 8
```

**Comment by Mailaender** [comment]
I can't reproduce the crash as it crashes for other reasons, but I think this is related to

```salt
SHOCKDROP:
	ParatroopersPower:
		DisplayBeacon: False
		DropItems: SHOK, SHOK
```

```lua
ShockDrop = Actor.Create("shockdrop", false, { Owner = Creeps })
```

**Comment by dwwwj222** [comment]
[nuclear03-debug.zip](https://github.com/OpenRA/OpenRA/files/7149477/nuclear03-debug.zip)

Hi, 

I have fixed all the errors in the map (attached), the map works fine without debug but as before if i disable the visibility checks at start it crashes the map with the same error message.

Might be something to do with the visibility but still happens after I fixed the visibility errors.

Thanks for any help

**Comment by Mailaender** [comment]
```
Exception of type `System.NullReferenceException`: Object reference not set to an instance of an object
  at OpenRA.Mods.Common.Traits.SupportPower.PlayLaunchSounds () [0x0007d] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Traits/SupportPowers/SupportPower.cs:190 
  at OpenRA.Mods.Common.Traits.ParatroopersPower+<>c__DisplayClass4_0.<SendParatroopers>b__3 (OpenRA.World w) [0x0000d] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Traits/SupportPowers/ParatroopersPower.cs:195 
  at OpenRA.World.Tick () [0x001ba] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/World.cs:445 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x001bc] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Game.cs:620 
  at OpenRA.Game.LogicTick () [0x0003e] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Game.cs:646 
  at OpenRA.Game.Loop () [0x000f1] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Game.cs:811 
  at OpenRA.Game.Run () [0x0003c] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Game.cs:852 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00010] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Game.cs:274 
  at OpenRA.Launcher.Program.Main (System.String[] args) [0x00044] in /home/matthias/Entwicklung/OpenRA/OpenRA.Launcher/Program.cs:30
```

**Comment by PunkPun** [comment]
Fixed by #19872
